codegen/render/kube_multicluster_test.go

```diff
@@ -0,0 +1,403 @@
+package render_test
+
+import (
+	"context"
+	"os"
+	"sync"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/solo-io/go-utils/randutils"
+	. "github.com/solo-io/skv2/codegen/test/api/things.test.io/v1"
+	mc_client "github.com/solo-io/skv2/pkg/multicluster/client"
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/solo-io/go-utils/kubeutils"
+	kubehelp "github.com/solo-io/go-utils/testutils/kube"
+	"github.com/solo-io/skv2/codegen/test/api/things.test.io/v1/controller"
+	"github.com/solo-io/skv2/pkg/multicluster"
+	"github.com/solo-io/skv2/pkg/multicluster/kubeconfig"
+	"github.com/solo-io/skv2/pkg/multicluster/watch"
+	"github.com/solo-io/skv2/pkg/reconcile"
+	"github.com/solo-io/skv2/test"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	zaputil "sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+func WithRemoteClusterContextDescribe(text string, body func()) bool {
+	if os.Getenv("REMOTE_CLUSTER_CONTEXT") == "" {
+		return PDescribe("[This test depends on a second cluster with context REMOTE_CLUSTER_CONTEXT] "+text, body)
+	}
+	return Describe(text, body)
+}
+
+var _ = WithRemoteClusterContextDescribe("Multicluster", func() {
+	var (
+		ns            string
+		clientSet     Clientset
+		logLevel      = zap.NewAtomicLevel()
+		ctx           = context.TODO()
+		remoteContext = os.Getenv("REMOTE_CLUSTER_CONTEXT")
+	)
+
+	BeforeEach(func() {
+		logLevel.SetLevel(zap.DebugLevel)
+		log.SetLogger(zaputil.New(
+			zaputil.Level(&logLevel),
+		))
```

Comment:
Is this just to silence the logger?

Reply:
it sets logs to debug. this is a copy/paste from our other kube tests in this package, but it helped a lot with debugging the multicluster watcher becuase i could see which controllers were being started and stopped over the course of a test